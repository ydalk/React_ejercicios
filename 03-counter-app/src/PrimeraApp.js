//import React, {Fragment} from 'react';
import React from 'react';
import PropTypes from 'prop-types'


//puede tener varios argumentos
const PrimeraApp = ({saludo, subtitulo}) => {
    
    //no imprime booleano ni objetos
    //const saludo = 'Hola Mundo';

    return (
        // <> equivale al colocar un <Fragment></Fragment>
        /*para mostrar objetos debo usar la etiqueta <pre></pre> y 
        pasarle un objketo JSON.stringify 
         */

       <>
            <h1> { saludo } </h1>
            {/* <pre> { JSON.stringify( saludo, null, 3)} </pre> */}
            <p>{ subtitulo }</p>
       </>
               
    )
    
}

//valida que se envie un props de tipo string en el componente
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

//crear propiedades por defecto
PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}

//permite que se use el componente en otros lugares
export default PrimeraApp;