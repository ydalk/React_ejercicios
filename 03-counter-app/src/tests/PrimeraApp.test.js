import React from 'react';
import { shallow } from 'enzyme'

import PrimeraApp from "../PrimeraApp";

describe('Pruebas en <Primera App />', () => {


    // test('debe mostrar el mensaje "Hola soy Caro"', () => {

    //     const saludo = "Hola soy Caro";

    //     //const wrapper = render(<PrimeraApp saludo = {saludo} />);
    //     //wrapper.getByText(saludo);

    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);

    //     expect( getByText(saludo) ).toBeInTheDocument();
    // })

    test('debe mostrar <primeraApp /> correctamente', () => {

        const saludo = 'Hola soy Ari';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />)

        expect(wrapper).toMatchSnapshot();

    });
    //sirve para hacer evlluaciones en el html
    test('debe mostrar el subtitulo enviado por props', () => {

        const saludo = 'Hola soy Ari';
        const subtitulo = 'Soy el subtitulo'

        const wrapper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />)

        const textoParrafo = wrapper.find('p').text();
        //console.log(textoParrafo);

        expect(textoParrafo).toBe(subtitulo);
    });



})
