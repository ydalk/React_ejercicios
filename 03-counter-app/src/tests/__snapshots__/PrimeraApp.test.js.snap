// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Primera App /> debe mostrar <primeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola soy Ari
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
